docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+      return self._tables
+
+3. Define a reading and a writting method for your new component.
+
+.. code-block:: python
+
+  def read_tables(self, **kwargs):
+    """Read table files at <root> and parse to dict of dataframes"""
+    if not self._write:
+      self._tables = dict()  # start fresh in read-only mode
+
+    self.logger.info("Reading model table files.")
+    fns = glob.glob(join(self.root, f"*.csv"))
+    if len(fns) > 0:
+      for fn in fns:
+        name = basename(fn).split(".")[0]
+        tbl = pd.read_csv(fn)
+        self.set_tables(tbl, name=name)
+
+  def write_tables(self):
+    """Write tables at <root>."""
+    self._assert_write_mode
+    if len(self.tables) > 0:
+      self.logger.info("Writing table files.")
+      for name in self.tables:
+        fn_out = join(self.root, f"{name}.csv")
+        self.tables[name].to_csv(fn_out, sep=",", index=False, header=True)
+
+4. Define a set function in order to add or update data in your component.
+
+.. code-block:: python
+
+  def set_tables(self, df: pd.DataFrame, name:str):
+    """Add table <pandas.DataFrame> to model."""
+    if not (isinstance(df, pd.DataFrame) or isinstance(df, pd.Series)):
+      raise ValueError("df type not recognized, should be pandas.DataFrame or pandas.Series.")
+    if name in self._tables:
+      if not self._write:
+        raise IOError(f"Cannot overwrite table {name} in read-only mode")
+      elif self._read:
+        self.logger.warning(f"Overwriting table: {name}")
+    self._tables[name] = df
+
+.. _plugin_workflows:
+
+Data processing: setup methods and workflows
+--------------------------------------------
+Now let's focus on the wheel part of the schematic diagram or how to add data to your model using HydroMT, which is what it's all about!
+Nothing here is really mandatory but more about conventions or tips also from experience with other plugins.
+
+But in general (scheme below):
+
+- To build a model we specify ``setup_<>`` methods which transform raw input data to a specific model variable(s), for instance
+  the ``setup_landuse`` method in HydroMT-Wflow to map landuse classes to associated Wflow parameter grids which are part of
+  the ``grid`` component.
+- All public model methods may only contain arguments which require one of the following basic python types: string, numeric
+  integer and float, boolean, None, list and dict types. This requirement makes it possible to expose these methods and their arguments
+  via the hydromt configuration in YAML format.
+- Data is exposed to each model method through the ``Model.data_catalog`` attribute which is an instance of the
+  :py:class:`hydromt.DataCatalog`. Data of :ref:`supported data types <data_types>` is provided to model methods
+  by arguments which end with ``_fn`` (short for filename) which refer to a source in the data catalog
+  based on the source name or a file based on the (relative) path to the file. Within a model method the data is read
```

Comment:
I would suggest to skip the **_fn** convention. We don't use this anymore in SfincsModel as many users, use the python interface and pass objects which are not read by the data adapters.